gem/gem.py

```diff
@@ -149,18 +149,20 @@ def array(self):
 class Literal(Constant):
     """Tensor-valued constant"""
 
-    __slots__ = ('array',)
+    __slots__ = ('array', 'name')
     __front__ = ('array',)
+    __back__ = ('name',)
 
-    def __new__(cls, array):
+    def __new__(cls, array, name=None):
```

Comment:
But the gem dag and code generation don't care about node names like this. So no need.